packages/world/src/systems/FarmingSystem.sol

```diff
@@ -35,150 +33,8 @@ contract FarmingSystem is System {
     (, ObjectTypeId toolType) = InventoryUtils.useTool(caller, toolSlot, type(uint128).max);
     require(toolType.isHoe(), "Must equip a hoe");
 
-    FarmingLib._processEnergyReduction(caller);
-
-    ObjectType._set(farmland, ObjectTypes.Farmland);
-  }
-
-  function growSeed(EntityId caller, Vec3 coord) external {
-    caller.activate();
-    // TODO: should we do proximity checks?
-
-    (EntityId seed, ObjectTypeId objectTypeId) = getOrCreateEntityAt(coord);
-    require(objectTypeId.isSeed(), "Not a seed");
-
-    require(SeedGrowth._getFullyGrownAt(seed) <= block.timestamp, "Seed cannot be grown yet");
-
-    // When a seed grows, it's removed from circulation
-    // We only update ResourceCount since seeds don't participate in respawning (no need to track positions
-    uint256 seedCount = ResourceCount._get(objectTypeId);
-    // This should never happen if there are seeds in the world obtained from drops
-    require(seedCount > 0, "Not enough seeds in circulation");
-    ResourceCount._set(objectTypeId, seedCount - 1);
-
-    if (objectTypeId.isCropSeed()) {
-      // Turn wet farmland to regular farmland if mining a seed or crop
-      (EntityId below, ObjectTypeId belowTypeId) = getOrCreateEntityAt(coord - vec3(0, 1, 0));
-      // Sanity check
-      if (belowTypeId == ObjectTypes.WetFarmland) {
-        ObjectType._set(below, ObjectTypes.Farmland);
-      }
-
-      ObjectType._set(seed, objectTypeId.getCrop());
-    } else if (objectTypeId.isTreeSeed()) {
-      TreeData memory treeData = objectTypeId.getTreeData();
-
-      // Grow the tree (replace the seed with the trunk and add blocks)
-      (uint32 trunkHeight, uint32 leaves) = FarmingLib._growTree(seed, coord, treeData);
-
-      // Seed energy is the sum of the energy of all the blocks of the tree
-      uint128 seedEnergy = ObjectTypeMetadata._getEnergy(objectTypeId);
-
-      uint128 trunkEnergy = trunkHeight * ObjectTypeMetadata._getEnergy(treeData.logType);
-      uint128 leafEnergy = leaves * ObjectTypeMetadata._getEnergy(treeData.leafType);
-
-      uint128 energyToReturn = seedEnergy - trunkEnergy - leafEnergy;
-      if (energyToReturn > 0) {
-        addEnergyToLocalPool(coord, energyToReturn);
-      }
-    }
-  }
-}
-
-library FarmingLib {
-  function _processEnergyReduction(EntityId caller) public {
     transferEnergyToPool(caller, TILL_ENERGY_COST);
-  }
-
-  function _growTree(EntityId seed, Vec3 baseCoord, TreeData memory treeData) public returns (uint32, uint32) {
```

Comment:
Was a bunch of this code modified or just moved to NatureSystem?

Reply:
Yeah a lot changed, now we operate over the predefined tree data instead of doing calculations at runtime